client/state/plans/actions.js

```diff
@@ -1,22 +1,35 @@
 /**
+ * External dependencies
+ */
+import findIndex from 'lodash/findIndex';
+
+/**
  * Internal dependencies
  */
 import wpcom from 'lib/wp';
-
 import {
 	PLANS_RECEIVE,
 	PLANS_REQUEST,
 	PLANS_REQUEST_SUCCESS,
 	PLANS_REQUEST_FAILURE
 } from '../action-types';
 
+import { PLAN_FREE, PLAN_PERSONAL } from 'lib/plans/constants';
+import personalPlan from 'lib/plans/personal-plan';
 /**
  * Action creator function: RECEIVE
  *
  * @param {Array} plans - WordPress.com plans list
  * @return {Object} action object
  */
 export const plansReceiveAction = plans => {
+	const freePlanIndex = findIndex( plans, plan => plan.product_slug === PLAN_FREE );
+	const personalPlanIndex = findIndex( plans, plan => plan.product_slug === PLAN_PERSONAL );
+
+	if ( personalPlanIndex < 0 ) {
+		plans.splice( freePlanIndex + 1, 0, personalPlan );
```

Comment:
Watch out for unexpected mutation! This modifies `plans` - see http://jsbin.com/xisanufimu/edit?js,console

It looks like you are trying to insert the personal plan into the list of plans just after the free one. If that's the case, how about an immutable approach?

```js
const freePlanIndex = findIndex( … );
const hasPersonalPlan = plans.some( ( { product_slug } ) => PLAN_PERSONAL === product_slug );

const newPlans = ! hasPersonalPlan
	? [ ...plans.slice( 0, freePlanIndex ), personalPlan, ...plans.slice( freePlanIndex ) ]
	: plans;

return { type: PLANS_RECEIVE, plans };
```